xcp_d/interfaces/connectivity.py

```diff
@@ -119,23 +140,321 @@ def _run_interface(self, runtime):
 
         masker = NiftiLabelsMasker(
             labels_img=atlas,
+            labels=node_labels,
             mask_img=mask,
             smoothing_fwhm=None,
             standardize=False,
+            resampling_target=None,  # they should be in the same space/resolution already
         )
 
         # Use nilearn for time_series
-        time_series = masker.fit_transform(filtered_file)
+        timeseries_arr = masker.fit_transform(filtered_file)
+        assert timeseries_arr.shape[1] == n_found_nodes
 
         # Apply the coverage mask
-        time_series[:, coverage_thresholded] = 0
+        timeseries_arr[:, coverage_thresholded] = np.nan
+
+        # Region indices in the atlas may not be sequential, so we map them to sequential ints.
+        seq_mapper = {idx: i for i, idx in enumerate(node_labels_df.index.tolist())}
```

Comment:
is there a guarantee that they're already in order? Would it make sense to have `enumerate(sorted(node_labels_df.index.tolist()))`

Reply:
That's a good point. I just added something to work around this. I implemented it a little differently, but it should accomplish the same thing.